packages/esm/core/src/Engines/WebGL/Extensions/rawTexture/engine.rawTexture.webgl.ts

```diff
@@ -0,0 +1,597 @@
+import { InternalTexture, InternalTextureSource } from "@babylonjs/core/Materials/Textures/internalTexture.js";
+import type { Nullable } from "@babylonjs/core/types.js";
+import type { IWebGLEnginePublic, WebGLEngineStateFull } from "../../engine.webgl.js";
+import {
+    _bindTextureDirectly,
+    _getInternalFormat,
+    _getRGBABufferInternalSizedFormat,
+    _getSamplingParameters,
+    _getUseSRGBBuffer,
+    _getWebGLTextureType,
+    _unpackFlipY,
+} from "../../engine.webgl.js";
+import { Logger } from "@babylonjs/core/Misc/logger.js";
+import { Tools } from "@babylonjs/core/Misc/tools.js";
+import type { Scene } from "@babylonjs/core/scene.js";
+import type { IWebRequest } from "@babylonjs/core/Misc/interfaces/iWebRequest.js";
+import type { IRawTextureEngineExtension } from "../../../Extensions/rawTexture/engine.rawTexture.base.js";
+import { augmentEngineState } from "../../../engine.adapters.js";
+import { Constants } from "../../../engine.constants.js";
+import { _loadFile } from "../../../engine.tools.js";
+
+export const updateRawTexture: IRawTextureEngineExtension["updateRawTexture"] = function (
+    engineState: IWebGLEnginePublic,
+    texture: Nullable<InternalTexture>,
+    data: Nullable<ArrayBufferView>,
+    format: number,
+    invertY: boolean,
+    compression: Nullable<string> = null,
+    type: number = Constants.TEXTURETYPE_UNSIGNED_INT,
+    useSRGBBuffer: boolean = false
+): void {
+    if (!texture) {
+        return;
+    }
+    const fes = engineState as WebGLEngineStateFull;
+    // Babylon's internalSizedFomat but gl's texImage2D internalFormat
+    const internalSizedFomat = _getRGBABufferInternalSizedFormat(fes, type, format, useSRGBBuffer);
+
+    // Babylon's internalFormat but gl's texImage2D format
+    const internalFormat = _getInternalFormat(fes, format);
+    const textureType = _getWebGLTextureType(fes, type);
+    _bindTextureDirectly(fes, fes._gl.TEXTURE_2D, texture, true);
+    _unpackFlipY(fes, invertY === undefined ? true : invertY ? true : false);

```

Comment:
Why not `_unpackFlipY(fes, invertY === undefined ? true : invertY);`  or even `_unpackFlipY(fes, !!invertY);`?